pages/emotion/[id]/edit.js

```diff
@@ -0,0 +1,26 @@
+import EmotionForm from "@/components/EmotionForm";
+import Heading from "@/components/Heading";
+import { useRouter } from "next/router";
+
+export default function EditPage({ objects, onUpdateEmotion }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const existingEmotion = objects.find((object) => object.id === id);
+  if (!existingEmotion) {
+    return <p>Loading...</p>;
+  }
+
+  return (
+    <>
+      <Heading>Edit Emotion</Heading>
+      <EmotionForm
+        defaultValue={existingEmotion}
+        onSubmit={(data) => {
+          onUpdateEmotion({ ...data, id });
+          router.push(`/emotion/${id}`);
```

Comment:
The logic of where the user should be directed after submitting belongs in the `handleSubmit` function in the form, I think. 🤔  I would define two different functions in the `_app` (`handleEdit` and `handleCreate` or something like that) and pass them on accordingly. So here just ```onSubmit={onUpdateEmotion}```, but define the values ​​that are passed on in the `form`. 🤔 Please think about it again.

Reply:
We did that refactoring and created the handler functions inside the _app.js and passed on to the form etc.